### PR TITLE
Rewrite scenarios for BookMonograph and BookAnthology

### PR DESCRIPTION
**Pull request by axthosarouris**
Update scenarios for BookMonograph and BookAnthology for scenarios to include both types and also for scenarios to be more understandable for people that are not involved in the details of the mapping process (code)

### CODE REVIEW

File: cristin-import/src/test/java/cucumber/BookFeatures.java [hunk 1]
```diff
@@ -58,10 +57,10 @@ public void theBookReportHasAEqualTo(String numberOfPages) {
...
     public void theNvaResourceHasAPublicationContextWithNumberOfPagesEqualTo(String expectedNumberOfPages) {
         PublicationInstance<?> context = scenarioContext.getNvaEntry()
-                                             .getEntityDescription()
-                                             .getReference()
-                                             .getPublicationInstance();
-        BookMonograph book = (BookMonograph) context;
+            .getEntityDescription()
+            .getReference()
+            .getPublicationInstance();
+        PeerReviewedMonograph book = (PeerReviewedMonograph) context;
```
Comment (axthosarouris) [inline cristin-import/src/test/java/cucumber/BookFeatures.java:63]: Necessary whey checking for both BookMonograph and BookAnthology

File: cristin-import/src/test/resources/features/BookFeatures.feature [hunk 2]
```diff
@@ -0,0 +1,102 @@
+Feature: Book conversion features
```
Comment (axthosarouris) [inline cristin-import/src/test/resources/features/BookFeatures.feature:1]: All rules from the BookAnthology and the BookMonograph were moved here. Rules that applied to both types were adapted in order for both types to be checked

Comment (Eirik-Eide-Saur) [reply cristin-import/src/test/resources/features/BookFeatures.feature:1]: Should this file be called BookRules to match the rest of the feature files?

File: cristin-import/src/test/resources/features/BookFeatures.feature [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    When the Cristin Result is converted to an NVA Resource
+    Then the NVA Resource has a PublicationContext with publisher equal to "House of Publishing"
+    Examples:
+      | secondaryCategory |
+      | MONOGRAFI         |
+      | ANTOLOGI          |
+
+
+  Scenario Outline: NPI subject heading is copied from Cristin Result as is.
+  the Cristin Entry's Book report subjectField "subjectFieldCode" entry
```
Comment (Eirik-Eide-Saur) [inline cristin-import/src/test/resources/features/BookFeatures.feature:69]: Should this be something along the lines of "The value is stored in the Cristin Entry's Book report subjectField "subjectFieldCode" entry"?

Comment (axthosarouris) [reply cristin-import/src/test/resources/features/BookFeatures.feature:69]: I think I will remove the title completely.

**Review by Eirik-Eide-Saur — APPROVED** [review]

**Pull request merged**